pkg/istio-agent/health/health_check.go

```diff
@@ -152,7 +145,6 @@ func (w *WorkloadHealthChecker) PerformApplicationHealthCheck(callback func(*Pro
 			if numSuccess == w.config.SuccessThresh && !lastStateHealthy {
 				healthCheckLog.Info("success threshold hit, marking as healthy")
 				callback(&ProbeEvent{Healthy: true})
-				numSuccess = 0
```

Comment:
Why is this removed? This means `numSuccess == w.config.SuccessThresh` will happen exactly 1 one time, and going from pass -> fail -> pass will not trigger an update?

Reply:
nope, the other branch will set it to 0 when it is unhealthy